lib/insights/api/common/audit_log.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require_relative 'audit_log/formatter'
+require_relative 'audit_log/middleware'
+
+module Insights::API::Common
+  # Audit Logger into selected logger, but primarily to CloudWatch
+  class AuditLog
+    class << self
+      def logger
+        @logger ||= init_logger
+      end
+
+      def logger=(logger)
+        @logger = init_logger(logger)
+      end
+
+      def setup(logger = nil)
+        self.logger = logger
+      end
+
+      def with_account(account_number)
+        original = Thread.current[:audit_account_number]
+        Thread.current[:audit_account_number] = account_number
+        return unless block_given?
+
+        yield
+        Thread.current[:audit_account_number] = original
+      end
+
+      private
+
+      def init_logger(logger = nil)
+        logger ||= Logger.new($stdout)
+        logger.level = Logger::INFO
```

Comment:
in topological/sources we are using ENV['LOG_LEVEL'] for ruby and ENV['RAILS_LOG_LEVEL'] for rails. 
what about `logger.level= ENV[...] || Logger::INFO`

Reply:
The level is hardcoded, as this is not going to be configurable. The idea is to use info/warn/error for audit logs. The end result might not even use and/or format the level. The difference would be only whether it is a failure/success.